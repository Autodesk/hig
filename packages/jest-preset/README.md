# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
