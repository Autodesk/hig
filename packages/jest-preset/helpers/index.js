/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
