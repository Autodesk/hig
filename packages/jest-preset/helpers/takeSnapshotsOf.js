const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
