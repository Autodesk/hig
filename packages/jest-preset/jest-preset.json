{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
