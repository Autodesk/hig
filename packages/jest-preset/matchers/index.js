require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
