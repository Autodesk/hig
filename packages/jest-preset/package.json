{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
