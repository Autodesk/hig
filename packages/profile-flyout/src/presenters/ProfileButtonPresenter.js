import React from "react";
import PropTypes from "prop-types";
import Avatar, { sizes } from "@hig/avatar";
import { css } from "emotion";
import stylesheet from "./stylesheet";

export default function ProfileButtonPresenter({
  avatarImage,
  avatarName,
  onClick,
  stylesheet: customStylesheet
}) {
  const styles = stylesheet({ stylesheet: customStylesheet });

  return (
    <button
      type="button"
      className={css(styles.flyoutButton)}
      onClick={onClick}
    >
      <Avatar name={avatarName} image={avatarImage} size={sizes.MEDIUM_32} />
    </button>
  );
}

ProfileButtonPresenter.propTypes = {
  /** Url to a profile image */
  avatarImage: PropTypes.string,
  /** The name that will converted into initials, and displayed when an image isn't available */
  avatarName: PropTypes.string,
  /** Callback when the flyout is opened */
  onClick: PropTypes.func,
  /** Function to modify the component's styles */
  stylesheet: PropTypes.func
};
