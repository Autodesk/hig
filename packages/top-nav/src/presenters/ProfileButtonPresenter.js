import React from "react";
import PropTypes from "prop-types";
import Avatar, { sizes } from "@hig/avatar";

export default function ProfileButtonPresenter({
  avatarImage,
  avatarName,
  onClick
}) {
  return (
    <button type="button" onClick={onClick}>
      <Avatar name={avatarName} image={avatarImage} size={sizes.MEDIUM_32} />
    </button>
  );
}

ProfileButtonPresenter.propTypes = {
  /** Url to a profile image */
  avatarImage: PropTypes.string,
  /** The name that will converted into initials, and displayed when an image isn't available */
  avatarName: PropTypes.string.isRequired,
  /** Callback when the flyout is opened */
  onClick: PropTypes.func
};
