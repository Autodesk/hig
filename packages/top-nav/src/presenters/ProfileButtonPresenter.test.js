import React from "react";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

import ProfileButtonPresenter from "./ProfileButtonPresenter";

describe("top-nav/presenters/ProfileButtonPresenter", () => {
  takeSnapshotsOf(ProfileButtonPresenter, [
    {
      desc: "renders with avatarName",
      props: {
        avatarName: "Wolfgang Pauli",
      },
    },
    {
      desc: "renders with children and avatarName",
      props: {
        avatarName: "Hans Hellmann",
        children: <div className="logo" />,
      },
    },
    {
      desc: "renders with all props",
      props: {
        avatarName: "Richard Feynman",
        avatarImage: "richard_phillips_feynman.png",
        children: <div className="logo" />,
        onClick: () => {},
      },
    },
  ]);
});
