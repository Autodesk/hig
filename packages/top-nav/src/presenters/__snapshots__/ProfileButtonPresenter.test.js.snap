// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`top-nav/presenters/ProfileButtonPresenter  1`] = `
.emotion-1 {
  background-color: #163b84;
  color: #ffffff;
  width: 32px;
  height: 32px;
  line-height: 32px;
  font-size: 14px;
  display: block;
  position: relative;
  margin: 0;
  overflow: hidden;
  border-radius: 50%;
  text-align: center;
}

.emotion-0 {
  width: 32px;
  height: 32px;
  font-family: ArtifaktElement,sans-serif;
}

<button
  onClick={undefined}
  type="button"
>
  <span
    aria-label="Avatar for Wolfgang Pauli"
    className="emotion-1"
    role="img"
  >
    <span
      aria-hidden="true"
      className="emotion-0"
    >
      WP
    </span>
  </span>
</button>
`;

exports[`top-nav/presenters/ProfileButtonPresenter  2`] = `
.emotion-0 {
  width: 32px;
  height: 32px;
  font-family: ArtifaktElement,sans-serif;
}

.emotion-1 {
  background-color: #5b5bc2;
  color: #ffffff;
  width: 32px;
  height: 32px;
  line-height: 32px;
  font-size: 14px;
  display: block;
  position: relative;
  margin: 0;
  overflow: hidden;
  border-radius: 50%;
  text-align: center;
}

<button
  onClick={undefined}
  type="button"
>
  <span
    aria-label="Avatar for Hans Hellmann"
    className="emotion-1"
    role="img"
  >
    <span
      aria-hidden="true"
      className="emotion-0"
    >
      HH
    </span>
  </span>
</button>
`;

exports[`top-nav/presenters/ProfileButtonPresenter  3`] = `
.emotion-2 {
  width: 32px;
  height: 32px;
  font-family: ArtifaktElement,sans-serif;
}

.emotion-3 {
  background-color: #5b5bc2;
  color: #ffffff;
  width: 32px;
  height: 32px;
  line-height: 32px;
  font-size: 14px;
  display: block;
  position: relative;
  margin: 0;
  overflow: hidden;
  border-radius: 50%;
  text-align: center;
}

.emotion-1 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 2;
  font-size: 14px;
}

.emotion-0 {
  border-radius: 50%;
  width: 32px;
  height: 32px;
}

<button
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="Avatar for Richard Feynman"
    className="emotion-3"
    role="img"
  >
    <span
      className="emotion-1"
    >
      <img
        alt="Avatar image of Richard Feynman"
        className="emotion-0"
        onError={[Function]}
        src="richard_phillips_feynman.png"
      />
    </span>
    <span
      aria-hidden="true"
      className="emotion-2"
    >
      RF
    </span>
  </span>
</button>
`;
